tests/integration/conftest.py

```diff
@@ -471,3 +471,112 @@ async def s3_integrator_app_backup_fixture(
     yield s3_integrator_app
     await model.remove_application(s3_integrator_app_name)
     await model.block_until(lambda: s3_integrator_app_name not in model.applications, timeout=60)
+
+
+@pytest.fixture(scope="function", name="s3_media_configuration")
+def s3_media_configuration_fixture(localstack_address: str) -> dict:
+    """Return the S3 configuration to use for media
+
+    Returns:
+        The S3 configuration as a dict
+    """
+    return {
+        "endpoint": f"http://{localstack_address}:4566",
+        "bucket": "synapse-media-bucket",
+        "path": "/media",
+        "region": "us-east-1",
+        "s3-uri-style": "path",
+    }
+
+
+@pytest.fixture(scope="module", name="s3_media_credentials")
+def s3_media_credentials_fixture(localstack_address: str) -> dict:
+    """Return the S3 AWS credentials to use for media
+
+    Returns:
+        The S3 credentials as a dict
+    """
+    return {
+        "access-key": token_hex(16),
+        "secret-key": token_hex(16),
+    }
+
+
+@pytest.fixture(scope="module", name="s3_media_integrator_name")
+def s3_media_integrator_name_fixture() -> str:
+    """Return the name of the s3 integrator application deployed for tests."""
+    return "s3-integrator-media"
+
+
+@pytest_asyncio.fixture(scope="function", name="s3_integrator_app_media")
+async def s3_integrator_app_media_fixture(
+    model: Model,
+    s3_media_configuration: dict,
+    s3_media_credentials: dict,
+    s3_media_integrator_name: str,
+):
+    """Returns a s3-integrator app configured with backup parameters."""
+    s3_integrator_app_name = s3_media_integrator_name
+    s3_integrator_app = await model.deploy(
+        "s3-integrator",
+        application_name=s3_integrator_app_name,
+        channel="latest/edge",
+        config=s3_media_configuration,
+    )
+    await model.wait_for_idle(apps=[s3_integrator_app_name], idle_period=5, status="blocked")
+    action_sync_s3_credentials: Action = await s3_integrator_app.units[0].run_action(
+        "sync-s3-credentials",
+        **s3_media_credentials,
+    )
+    await action_sync_s3_credentials.wait()
+    await model.wait_for_idle(apps=[s3_integrator_app.name], status="active")
+
+    yield s3_integrator_app
+    await model.remove_application(s3_integrator_app_name)
```

Comment:
Minor: is there a reason to refer to s3 integrator app name using different variables?

Reply:
will make this consistent :)